pytorch_lightning/trainer/training_io.py

```diff
@@ -307,8 +307,8 @@ def restore(self, checkpoint_path, on_gpu):
     def dump_checkpoint(self):
 
         checkpoint = {
-            'epoch': self.current_epoch,
-            'global_step': self.global_step
+            'epoch': self.current_epoch + 1,
```

Comment:
The problem is that checkpoint can be saved not only at the end of the training epoch. For example, if you set `val_check_interval=0.1` and after `0.15` of the training batches the training was interrupted you will continue from the second epoch whereas only `10%` of the first epoch actually was processed.

Reply:
That's a good point, I'll look into better ways of dealing with this

Reply:
Looking at it, I think resuming in such a way is probably a new feature, which should go in a new PR and have more discussion there. I've updated the test to allow for testing mid-epoch check pointing / resuming but commented out the checkpoints from mid-epochs so as to pass with the current resume method. When it is properly implemented we can use the full test. 

For now I've added a warning if you load a mid-epoch checkpoint to alert the user that it will be unreliable to resume.